### ISSUE TITLE
Disable Maniaplanet.StartMap_End for TM2020

### ISSUE THREAD

**Issue by jonthekiller** [ManiaControl/ManiaControl#275]
It seems that the Maniaplanet.StartMap_End callback should not be take into account (probably also the Maniaplanet.EndMap_End) as for me we always receive the ManiaPlanet.BeginMap and ManiaPlanet.EndMap. As of today, in some conditions (for example during a warm-up and do skip map) we can receive twice the same callback (StartMap_End, EndMap, and after BeginMap) and give some bad behaviour in some plugins using these callbacks. I would recommend to disable this two for TM2020. I haven't check all scenarii to see if in some special cases, we need them but it seems not.
https://github.com/ManiaControl/ManiaControl/blob/24558b1982050ad9dc7544afbbfa53bfe79440e3/core/Callbacks/LibXmlRpcCallbacks.php#L64

**Comment by kremsy** [comment]
Please be really careful with these start/end callbacks. Especially in
Shootmania all of these are heavily used and the order is really
important. Changing some handling to a later or earlier callback could
break a lot... 

Am 2020-08-02 18:17, schrieb jonthekiller:

> It seems that the Maniaplanet.StartMap_End callback should not be take into account (probably also the Maniaplanet.EndMap_End) as for me we always receive the ManiaPlanet.BeginMap and ManiaPlanet.EndMap. As of today, in some conditions (for example during a warm-up and do skip map) we can receive twice the same callback (StartMap_End, EndMap, and after BeginMap) and give some bad behaviour in some plugins using these callbacks. I would recommend to disable this two for TM2020. I haven't check all scenarii to see if in some special cases, we need them but it seems not.
> https://github.com/ManiaControl/ManiaControl/blob/24558b1982050ad9dc7544afbbfa53bfe79440e3/core/Callbacks/LibXmlRpcCallbacks.php#L64 
> 
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].

 

Links:
------
[1] https://github.com/ManiaControl/ManiaControl/issues/275
[2]

**Comment by jonthekiller** [comment]
Yes, I think it's due to the WarmUp phase that is not managed like in TM2 and can be important for gamemodes.
At least, for short term, we need to disable only for TM2020 but I haven't check the best way to do it yet.

**Comment by jonthekiller** [comment]
I can propose this modification: https://github.com/jonthekiller/ManiaControl/blob/jonthekiller-patch-13/core/Callbacks/LibXmlRpcCallbacks.php#L64

Something better can probably be done.
